v2/pkg/engine/resolve/fetch.go

```diff
@@ -291,23 +291,38 @@ type FetchConfiguration struct {
 	Input      string
 	Variables  Variables
 	DataSource DataSource
-	// RequiresParallelListItemFetch is used to indicate that the single fetches should be executed without batching
-	// When we have multiple fetches attached to the object - after post-processing of a plan we will get ParallelListItemFetch instead of ParallelFetch
+
+	// RequiresParallelListItemFetch indicates that the single fetches should be executed without batching.
+	// If we have multiple fetches attached to the object, then after post-processing of a plan
```

Comment:
this is for the objects under array path

used only for the introspection datasource rn